src/app/designationEditButton/designationEditButton.component.js

```diff
@@ -17,7 +17,7 @@ class DesignationEditButtonController {
 
   $onInit() {
     if(this.designationNumber && includes([Roles.identified, Roles.registered], this.sessionService.getRole())){
-      this.designationEditorService.getContent(this.designationNumber, this.campaignPage).then(() => {
+      this.designationEditorService.checkPermission(this.designationNumber, this.campaignPage).then(() => {
```

Comment:
You should add an error callback here as well. Have it $log to rollbar if an error occurs. Not sure if it needs to let the user know if an error happened.